src/v/utils/tests/delta_for_test.cc

```diff
@@ -195,4 +232,24 @@ BOOST_AUTO_TEST_CASE(random_walk_test_8) {
     static constexpr int test_size = 100000;
     static constexpr int max_delta = 100000;
     test_random_walk_roundtrip<int64_t>(test_size, max_delta);
+}
+
+BOOST_AUTO_TEST_CASE(test_compression_ratio) {
+    int test_size = 100000;
+    static constexpr uint64_t initial_value = 0;
+    static constexpr uint64_t min_step = 10000;
+    deltafor_encoder<uint64_t> enc_xor(initial_value);
+    deltafor_encoder<uint64_t, details::delta_delta<uint64_t>> enc_delta(
+      initial_value, details::delta_delta(min_step));
+    std::vector<std::pair<uint64_t, uint64_t>> deltas;
+    deltas.reserve(test_size);
+    for (int i = 0; i < test_size; i++) {
+        deltas.emplace_back(std::make_pair(min_step, min_step + 100));
+    }
+    populate_encoder(enc_xor, initial_value, deltas);
+    populate_encoder(enc_delta, initial_value, deltas);
+    BOOST_REQUIRE(
+      enc_xor.share().size_bytes() > enc_delta.share().size_bytes());
+    auto byte_per_val = test_size * 16;
```

Comment:
The naming is a bit confusing - am I correct that test size is measured in rows? So `test_size * 16` (or, better, `FOR_buffer_length`) is actually "number of values", and `byte_per_val` is actually 1? I.e. here we check that `delta_delta` encoder compresses the values to less then 1 byte per value?

Reply:
fixed that